Makefile

```diff
@@ -108,13 +106,13 @@ manifests-dev: cmd/clusterctl/examples/aws/provider-components-base-dev.yaml ##
 
 .PHONY: cmd/clusterctl/examples/aws/provider-components-base.yaml
 cmd/clusterctl/examples/aws/provider-components-base.yaml:
-	bazel build //cmd/clusterctl/examples/aws:provider-components-base $(BAZEL_DOCKER_ARGS)
+	bazel build //cmd/clusterctl/examples/aws:provider-components-base $(BAZEL_ARGS)
 	install bazel-genfiles/cmd/clusterctl/examples/aws/provider-components-base.yaml cmd/clusterctl/examples/aws
 
 .PHONY: cmd/clusterctl/examples/aws/provider-components-base-dev.yaml
 cmd/clusterctl/examples/aws/provider-components-base-dev.yaml:
-	bazel build //cmd/clusterctl/examples/aws:provider-components-base-dev $(BAZEL_DOCKER_ARGS)
-	install bazel-genfiles/cmd/clusterctl/examples/aws/provider-components-base-dev.yaml cmd/clusterctl/examples/aws
+	bazel build //cmd/clusterctl/examples/aws:provider-components-base $(BAZEL_ARGS) --define=PULL_POLICY=Always --define=REGISTRY=gcr.io/$(shell gcloud config get-value project)
```

Comment:
Are we overriding anything other than PULL_POLICY and REGISTRY here? Any reason this needs to be a separate target from provider-components-base.yaml?

Reply:
I wanted to keep the Make commands the same.

Correct me if I'm wrong, but what you're suggesting is:

```
target:
  bazel build thing --define=PULL_POLICY=$PULL_POLICY
```

to combine the -dev target and this target.

TBH I agree here and would prefer development builds to be default. That would mean the development workflow would be

```
make docker-push
```

and the release workflow would be

```
make docker-push PULL_POLICY=IfNotPreset REGISTRY=gcr.io/cluster-api-provider-aws
```

Does that sound reasonable?

Reply:
+1 to making the dev workflow the default and overriding (in a documented way) for the release workflow :)